base_layer/wallet/tests/transaction_service/service.rs

```diff
@@ -2841,10 +2837,9 @@ fn test_direct_vs_saf_send_of_tx_reply_and_finalize() {
 
     let _ = alice_outbound_service.wait_call_count(1, Duration::from_secs(60));
     let _ = alice_outbound_service.pop_call().unwrap();
-    for _ in 0..3 {
-        assert_eq!(alice_outbound_service.call_count(), 0, "Should be no more calls");
-        runtime.block_on(async { delay_for(Duration::from_secs(5)).await });
-    }
+
+    runtime.block_on(async { delay_for(Duration::from_secs(5)).await });
```

Comment:
Might want to use `async_assert_eventually!` i.e poll `alice_outbound_service.call_count()` every interval until it equals the expected value or times out 

```rust
runtime.block_on(async move {
async_asssert_eventually!(
   alice_outbound_service.call_count(),
   expect = 0,
   max_attempts = 20,
   interval = Duration::from_millis(250),
);
});

Reply:
In this case doesn't this amount to exactly the same thing? The difference to the normal usage is that we are waiting for the negative case so this will wait 5 seconds just like the original code.

Reply:
Oh I see - I'd say there's probably no need to test a negative or some other way that this should be tested. But ok, in that case `async_assert_eventually!` is not correct.